private/bufpkg/bufplugin/bufplugindocker/docker.go

```diff
@@ -169,11 +187,32 @@ func (d *dockerAPIClient) Build(ctx context.Context, dockerfile io.Reader, plugi
 		}()
 		scanner := bufio.NewScanner(response.Body)
 		for scanner.Scan() {
-			d.logger.Debug(scanner.Text())
 			var message jsonmessage.JSONMessage
-			if err := json.Unmarshal([]byte(scanner.Text()), &message); err == nil && message.Error != nil {
+			if err := json.Unmarshal([]byte(scanner.Text()), &message); err != nil {
+				d.logger.Debug(scanner.Text())
+				continue
+			}
+			if message.Error != nil {
 				return message.Error
 			}
+			handled := false
+			if message.ID == "moby.buildkit.trace" && message.Aux != nil {
```

Comment:
This isn't pretty - messages are double quoted base64 strings which need to be unmarshalled as a protobuf type. This does allow us to get a lot more debug information though when building to determine what might be failing.

Reply:
Oh, thank you do much for adding this!